agent/consul/leader_connect_ca_test.go

```diff
@@ -234,13 +236,14 @@ func initTestManager(t *testing.T, manager *CAManager, delegate *mockCAServerDel
 }
 
 func TestCAManager_Initialize(t *testing.T) {
-
 	conf := DefaultConfig()
 	conf.ConnectEnabled = true
 	conf.PrimaryDatacenter = "dc1"
 	conf.Datacenter = "dc2"
 	delegate := NewMockCAServerDelegate(t, conf)
+	delegate.secondaryIntermediate = delegate.primaryRoot.RootCert
```

Comment:
This is definitely weird, but it preserves the old behaviour of this test. I think we'll need to do a bit more work to make this mock a little more realistic.